extensions/fs/internal.m

```diff
@@ -1090,6 +1091,34 @@ static int fs_pathFromBookmark(lua_State *L) {
     return 1 ;
 }
 
+/// hs.fs.urlFromPath(path) -> string | nil
+/// Function
+/// Returns the encoded URL from a path.
+///
+/// Parameters:
+///  * path - The path
+///
+/// Returns:
+///  * A string or `nil` if path is invalid.
+static int fs_urlFromPath(lua_State *L) {
+    LuaSkin *skin = [LuaSkin sharedWithState:L] ;
+    [skin checkArgs:LS_TSTRING, LS_TBREAK] ;
+
+    NSString *filePath = [skin toNSObjectAtIndex:1];
+    char *absolutePath = realpath([filePath stringByExpandingTildeInPath].UTF8String, NULL);
```

Comment:
More curiosity then anything else... does this differ from NSString's stringByStandardizingPath (https://developer.apple.com/documentation/foundation/nsstring/1407194-stringbystandardizingpath?language=occ)? I've used the NSString method in a couple of places myself, so if there is a difference, we may need to standardize...

Reply:
I don't think [`stringByStandardizingPath`](https://developer.apple.com/documentation/foundation/nsstring/1407194-stringbystandardizingpath) will convert `~/Desktop` to `/Users/*USERNAME*/Desktop`?

Reply:
Pretty sure it does, but can't test it right now (according to the docs, the first step is `Expanding an initial tilde expression using stringByExpandingTildeInPath.`)

Reply:
As usual, you're absolutely right. I've updated - thanks!